files/scripts/syncd.sh

```diff
@@ -163,11 +163,21 @@ stop() {
         debug "${TYPE} shutdown syncd process ..."
         /usr/bin/docker exec -i syncd$DEV /usr/bin/syncd_request_shutdown --${TYPE}
 
-        # wait until syncd quits gracefully
-        while docker top syncd$DEV | grep -q /usr/bin/syncd; do
+        # wait until syncd quits gracefully or force syncd to exit after 
+        # waiting for 20 seconds
+        start_in_secs=$(date +%s)
```

Comment:
bash has a buildin variable ${SECONDS} for up seconds. You can use that instead.

Reply:
Great suggestion! Updated.